File: app/dbmodels/Budget.php
<?php

$table = array(
    'fields'        => array(),
    'oldfields'     => array(),
    'renamefields'  => array(),
    'indexes'       => array(),
    'oldindexes'    => array(),
    'relationships' => array(),
    'model'         => 'Budget',
    'menu'          => 'Management'
);
$table['fields'] = array(
    'id'           => DBModels::type('increments',
                                     array('visible' => false)),
    'name'          => DBModels::type('string'),
    'comment'       => DBModels::type('text'),
    'entity_id'     => DBModels::type('integer'), ///TODO create relation
    'is_recursive'  => DBModels::type('boolean',
                                      array('visible' => false)),
    'is_deleted'    => DBModels::type('boolean',
                                      array('visible' => false)),
    'begin_date'    => DBModels::type('date'),
    'end_date'      => DBModels::type('date'),
    'value'         => DBModels::type('decimal'),
    'is_template'   => DBModels::type('boolean',
                                      array('visible' => false)),
    'template_name' => DBModels::type('string',
                                      array('visible' => false)),

);

File: fusionglpi/database/budget.py
from fusionglpi import db


class Budget(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    entity_id = db.Column(db.Integer, db.ForeignKey('entity.id'))
    is_recursive = db.Column(db.Boolean, nullable=False, server_default="0")
    name = db.Column(db.String(255, None, True))
    comment = db.Column(db.Text)
    is_deleted = db.Column(db.Boolean, nullable=False, server_default="0")
    begin_date = db.Column(db.Date)
    end_date = db.Column(db.Date)
    value = db.Column(db.Float, nullable=False, server_default="0")
    is_template = db.Column(db.Boolean, nullable=False, server_default="0")
    template_name = db.Column(db.String(255, None, True))

    infocoms = db.relationship('Infocom', backref='budget')
